src/item_tname.h

```diff
@@ -0,0 +1,188 @@
+#pragma once
+#ifndef CATA_SRC_ITEM_TNAME_H
+#define CATA_SRC_ITEM_TNAME_H
+
+#include <cstddef>
+#include <string>
+#include <unordered_map>
+
+#include "enum_bitset.h"
+#include "enum_traits.h"
+
+class item;
+
+namespace tname
```

Comment:
Can you comment what tname and fsegment etc are supposed to be? The short name is not intuitive to me.

Reply:
I think I've addressed everything, though I'm bad with comments so they might still be poor.